code/datums/trading/ai.dm

```diff
@@ -27,24 +27,66 @@ They sell generic supplies and ask for generic supplies.
 	)
 	possible_wanted_items = list(
 		/obj/item/device/                        = TRADER_SUBTYPES_ONLY,
+		/obj/item/device/kit					 = TRADER_BLACKLIST_ALL, // They're impossible to make or get outside of mining, just annoys traders.
 		/obj/item/device/assembly                = TRADER_BLACKLIST_ALL,
 		/obj/item/device/assembly_holder         = TRADER_BLACKLIST_ALL,
-		/obj/item/device/encryptionkey/syndicate = TRADER_BLACKLIST,
+		/obj/item/device/encryptionkey			 = TRADER_BLACKLIST_ALL, //Why does ai want NT encryption keys?
 		/obj/item/device/radio                   = TRADER_BLACKLIST_ALL,
 		/obj/item/device/pda                     = TRADER_BLACKLIST_SUB,
-		/obj/item/device/uplink                  = TRADER_BLACKLIST
+		/obj/item/device/chameleon				 = TRADER_BLACKLIST, // Why should it want a chameleon projector
+		/obj/item/device/dociler				 = TRADER_BLACKLIST, //Item unobtaineable
+		/obj/item/device/flashlight/drone		 = TRADER_BLACKLIST, // No drone stuff
+		/obj/item/device/camera_bug				 = TRADER_BLACKLIST, // Traitor stuff
+		/obj/item/device/multitool				 = TRADER_BLACKLIST_SUB, // Hacktool, uplink and robo tool ban
+		/obj/item/device/modkit					 = TRADER_BLACKLIST_ALL, // No to modkits
+		/obj/item/device/pin_extractor			 = TRADER_BLACKLIST, // RD's tech
+		/obj/item/device/powersink				 = TRADER_BLACKLIST, // Traitor stuff
+		/obj/item/device/slime_scanner			 = TRADER_BLACKLIST, //If it was doing slime stuff, it already had this
+		/obj/item/device/spy_bug				 = TRADER_BLACKLIST, // Traitor stuff
+		/obj/item/device/spy_monitor			 = TRADER_BLACKLIST, // Traitor stuff
+		/obj/item/device/suit_cooling_unit/improved = TRADER_BLACKLIST, // Not on Aurora
+		/obj/item/device/taperecorder/cciaa		 = TRADER_BLACKLIST, // Admin item
+		/obj/item/device/batterer				 = TRADER_BLACKLIST, // Item too rare
+		/obj/item/device/contract_uplink		 = TRADER_BLACKLIST, // Traitor stuff
+		/obj/item/device/uplink                  = TRADER_BLACKLIST_ALL, // Traitor stuff
+		/obj/item/device/announcer				 = TRADER_BLACKLIST, // Rev item
+		/obj/item/device/ninja_uplink			 = TRADER_BLACKLIST, // Ninja item
+		/obj/item/device/onetankbomb			 = TRADER_BLACKLIST, // Not weapons trader
+		/obj/item/device/kinetic_analyzer		 = TRADER_BLACKLIST, // Not KA trader
+		/obj/item/device/camera					 = TRADER_BLACKLIST_SUB, // a lot of ai/drone/cyborg/fluff items
```

Comment:
Can Easily be acquired

Reply:
![image](https://user-images.githubusercontent.com/6859396/62105853-deddc500-b2a3-11e9-8020-63171011e36a.png)
Blacklist_sub orders the trader to accept the main item and not accept the sub items. In this case, the trader would accept a normal camera found on station, but not cameras in ai's inventory and found on drones and cyborgs as well as custom fluff items.

Reply:
This mainly prevents confusion during trading when you try to trade for a camera but it will never accept because it's asking for the camera in Ai's inventory.

Reply:
Good point